Automattic/jetpack
# Boost: Add garbage collection

Fixes #35053

## Proposed changes:

* Add garbage collection
* Remove old log files during garbage collection
* Introduce `Filesystem_Utils` for common filesystem operations. Not all filesystem methods are moved here. But, eventually they should be.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create some cache by visiting different pages of the site. 
* Use the filter `jetpack_boost_cache_duration` to to modify the cache ttl (in seconds).
* Make sure the cron job for garbage collection is ran.
* Make sure the garbage collection works.
* Make sure the old log files get deleted.
